programs/bpf_loader/src/syscalls.rs

```diff
@@ -2328,7 +2331,21 @@ impl<'a, 'b> SyscallInvokeSigned<'a, 'b> for SyscallInvokeSignedC<'a, 'b> {
     ) -> Result<Instruction, EbpfError<BpfError>> {
         let ix_c = translate_type::<SolInstruction>(memory_mapping, addr, loader_id)?;
 
-        check_instruction_size(ix_c.accounts_len, ix_c.data_len, invoke_context)?;
+        assert_eq!(
```

Comment:
Can we make these debug asserts or move them to a test?

Reply:
Done.